airflow/sensors/base.py

```diff
@@ -281,6 +282,16 @@ def run_duration() -> float:
         self.log.info("Success criteria met. Exiting.")
         return xcom_value
 
+    def resume_execution(self, next_method: str, next_kwargs: dict[str, Any] | None, context: Context):
+        try:
+            return super().resume_execution(next_method, next_kwargs, context)
+        except (AirflowException, TaskDeferralError) as e:
+            if self.soft_fail:
+                raise AirflowSkipException(str(e))
```

Comment:
```suggestion
                raise AirflowSkipException(str(e)) from e
```

(I think we do this in many places)